File: 백준/Bronze/10798. 세로읽기/세로읽기.java
import java.io.*;
import java.util.Arrays;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int maxLen = 0;
        char[][] board = new char[5][15];

        for (int i = 0; i < 5; i++) {
            String arr = br.readLine();
            maxLen = Math.max(maxLen, arr.length());
            for (int j = 0; j < arr.length(); j++) {
                board[i][j] = arr.charAt(j);
            }
        }

        StringBuilder sb = new StringBuilder();
        for (int j = 0; j < maxLen; j++) {
            for (int i = 0; i < 5; i++) {
                if (board[i][j] != '\0') {
                    sb.append(board[i][j]);
                }
            }
        }
        System.out.println(sb);
    }

}

File: 백준/Bronze/10798. 세로읽기/세로읽기.py
board = []
max_l = 0
for i in range(5):
    arr = list(input())
    max_l = max(max_l, len(arr))
    board.append(arr)

answer = ''
for j in range(max_l):
    for i in range(5):
        if len(board[i]) > j:
            answer += board[i][j]
print(answer)